src/origin.rs

```diff
@@ -39,8 +39,9 @@ pub enum Origin {
     /// A globally unique identifier
     Opaque(OpaqueOrigin),
 
-    /// Consists of the URL's scheme, host and port
-    Tuple(String, Host<String>, u16)
+    /// Consists of the URL's scheme, host and port, and an optional domain
+    /// https://html.spec.whatwg.org/multipage/browsers.html#concept-origin-tuple
```

Comment:
I suspect the html spec origin might diverge from the URL spec origin, this crate is URL spec only